components/gantry/oneaxis/oneaxis.go

```diff
@@ -160,130 +68,67 @@ func newOneAxis(ctx context.Context, deps registry.Dependencies, config config.C
-	}
-
-	if err := oAx.Home(ctx); err != nil {
-		return nil, err
+	oAx.model, err = oAx.createModel(conf.Axis)
+	if err != nil {
+		return oAx, err
 	}
 
 	return oAx, nil
 }
 
-func (g *oneAxis) Home(ctx context.Context) error {
-	ctx, done := g.opMgr.New(ctx)
-	defer done()
+func (g *oneAxis) createModel(axis r3.Vector) (referenceframe.Model, error) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	var errs error
+	m := referenceframe.NewSimpleModel("")
 
-	// Mapping one limit switch motor0->limsw0, motor1 ->limsw1, motor 2 -> limsw2
-	// Mapping two limit switch motor0->limSw0,limSw1; motor1->limSw2,limSw3; motor2->limSw4,limSw5
-	switch g.limitType {
-	// An axis with one limit switch will go till it hits the limit switch, encode that position as the
-	// zero position of the one-axis, and adds a second position limit based on the steps per length.
-	case limitOnePin:
-		err := g.homeOneLimSwitch(ctx)
-		if err != nil {
-			return err
-		}
-	// An axis with two limit switches will go till it hits the first limit switch, encode that position as the
-	// zero position of the one-axis, then go till it hits the second limit switch, then encode that position as the
-	// at-length position of the one-axis.
-	case limitTwoPin:
-		err := g.homeTwoLimSwitch(ctx)
-		if err != nil {
-			return err
-		}
-	// An axis with an encoder will encode the
-	case limitEncoder:
-		err := g.homeEncoder(ctx)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
+	f, err := referenceframe.NewStaticFrame(g.name, spatialmath.NewZeroPose())
+	errs = multierr.Combine(errs, err)
+	m.OrdTransforms = append(m.OrdTransforms, f)
 
-func (g *oneAxis) homeTwoLimSwitch(ctx context.Context) error {
-	positionA, err := g.testLimit(ctx, true)
-	if err != nil {
-		return err
-	}
+	f, err = referenceframe.NewTranslationalFrame(g.name, axis, referenceframe.Limit{Min: 0, Max: g.lengthMm})
+	errs = multierr.Combine(errs, err)
 
-	positionB, err := g.testLimit(ctx, false)
-	if err != nil {
-		return err
+	if errs != nil {
+		return nil, errs
 	}
 
-	g.logger.Debugf("positionA: %0.2f positionB: %0.2f", positionA, positionB)
+	m.OrdTransforms = append(m.OrdTransforms, f)
 
-	g.positionLimits = []float64{positionA, positionB}
-
-	// Go backwards so limit stops are not hit.
-	x := g.rotationalToLinear(0.8 * g.lengthMm)
-	err = g.motor.GoTo(ctx, g.rpm, x, nil)
-	if err != nil {
-		return err
-	}
-	return nil
+	return m, errs
 }
 
-func (g *oneAxis) homeOneLimSwitch(ctx context.Context) error {
-	// One pin always and only should go backwards.
-	positionA, err := g.testLimit(ctx, true)
-	if err != nil {
-		return err
-	}
-
-	revPerLength := g.lengthMm / g.mmPerRevolution
-	positionB := positionA + revPerLength
-
-	g.positionLimits = []float64{positionA, positionB}
-
-	return nil
+func (g *oneAxis) linearToRotational(positions float64) float64 {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	theRange := g.positionLimits[1] - g.positionLimits[0]
```

Comment:
Throughout this PR, we're assuming the positions are sorted smallest to biggest. That's probably a good assumption, but I wonder if we should put a comment somewhere explicitly pointing it out. Should it go in the homing function? or at the top of a file as a package-level comment? or maybe not at all. but it seems to me that anyone who hooks their encoder up backwards is going to have a bad time, because the first position will be larger than the second.

Reply:
made `g.theRange` and absolute value, which should allow us a correct range without caring about directionality. We will still always go backwards first. That would be a motor fix.